extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1049,11 +1049,7 @@ private void updatePartitionDataFromKafka()
       return;
     }
 
-    List<PartitionInfo> partitions = topics.get(ioConfig.getTopic());
-    if (partitions == null) {
-      log.warn("No such topic [%s] found, list of discovered topics [%s]", ioConfig.getTopic(), topics.keySet());
-    }
-    int numPartitions = (partitions != null ? partitions.size() : 0);
+    int numPartitions = partitions.size();
```

Comment:
Looks like `partitions` can be null. Should be `int numPartitions = (partitions != null ? partitions.size() : 0);`.

Reply:
I think the partitions can be null only when consumer.partitionsFor(ioConfig.getTopic()) throw exceptions. And the exception has been catched and the function get return. So when partitions is null, there is no chance run int numPartitions = partitions.size();, is it?

Reply:
This is the code of `KafkaConsumer.partitionsFor()`.

```java
    public List<PartitionInfo> partitionsFor(String topic) {
        acquire();
        try {
            Cluster cluster = this.metadata.fetch();
            List<PartitionInfo> parts = cluster.partitionsForTopic(topic);
            if (parts != null)
                return parts;

            Map<String, List<PartitionInfo>> topicMetadata = fetcher.getTopicMetadata(
                    new MetadataRequest.Builder(Collections.singletonList(topic)), requestTimeoutMs);
            return topicMetadata.get(topic);
        } finally {
            release();
        }
    }
```

Since `topicMetadata` is a map, `topicMetadata.get(topic);` can return null.

Reply:
Ah~ I didn't notice that. Thank you. I will fix it.